prow/tide/gerrit.go

```diff
@@ -326,83 +325,17 @@ func (p *GerritProvider) mergePRs(sp subpool, prs []CodeReviewCommon, _ *threadS
 	return merged, utilerrors.NewAggregate(errs)
 }
 
-// GetTideContextPolicy gets context policy defined by users + requirements from
-// prow jobs.
-// * Required job: has label "prow.k8s.io/gerrit-report-label", and the value of
-// this label is a required Gerrit Label, and trigger condition is met.
-// * Required when present: has label "prow.k8s.io/gerrit-report-label", and the
-// value of this label is a required Gerrit Label, trigger condition is not met,
-// and `run_before_merge` is true.
-// * Optional: others.
+// GetTideContextPolicy returns an empty config.TideContextPolicy struct.
+//
+// These information are only for determining whether a PR is ready for merge or
+// not, this in Gerrit is handled by Gerrit query filters, so this is not useful
+// for Gerrit.
 func (p *GerritProvider) GetTideContextPolicy(org, repo, branch string, baseSHAGetter config.RefGetter, crc *CodeReviewCommon) (contextChecker, error) {
-	pr := crc.Gerrit
-	if pr == nil {
-		return nil, errors.New("programmer error: crc.Gerrit cannot be nil for GerritProvider")
-	}
-
-	required := sets.NewString()
-	requiredIfPresent := sets.NewString()
-	optional := sets.NewString()
-
-	headSHAGetter := func() (string, error) {
-		return crc.HeadRefOID, nil
-	}
-	cloneURI := gerritsource.CloneURIFromOrgRepo(org, repo)
-	// Get presubmits from Config alone.
-	presubmits, err := p.GetPresubmits(cloneURI, baseSHAGetter, headSHAGetter)
-	if err != nil {
-		return nil, fmt.Errorf("failed getting presubmits: %v", err)
-	}
-
-	requireLabels := sets.NewString()
-	for l, info := range pr.Labels {
-		if !info.Optional {
-			requireLabels.Insert(l)
-		}
-	}
-
-	// generate required and optional entries for Prow Jobs
-	for _, pj := range presubmits {
-		if !pj.CouldRun(branch) {
-			continue
-		}
-
-		// jobs that produce required contexts and will
-		// always run should be required at all times.
-		// jobs with `RunBeforeMerge` are also required.
-		var isJobRequired bool
-		// jobs that trigger conditionally are required if present.
-		var isJobRequiredWhenPresent bool
-
-		if pj.RunBeforeMerge {
-			isJobRequiredWhenPresent = true
-		}
-		if val, ok := pj.Labels[kube.GerritReportLabel]; ok && requireLabels.Has(val) {
-			if pj.TriggersConditionally() {
-				isJobRequiredWhenPresent = true
-			} else {
-				isJobRequired = true
-			}
-		}
-
-		if isJobRequired {
-			required.Insert(pj.Context)
-		} else if isJobRequiredWhenPresent {
-			requiredIfPresent.Insert(pj.Context)
-		} else {
-			optional.Insert(pj.Context)
-		}
-	}
-
-	t := &config.TideContextPolicy{
-		RequiredContexts:          required.List(),
-		RequiredIfPresentContexts: requiredIfPresent.List(),
-		OptionalContexts:          optional.List(),
-	}
-	if err := t.Validate(); err != nil {
-		return t, err
-	}
-	return t, nil
+	return &config.TideContextPolicy{
+		RequiredContexts:          make([]string, 0),
+		RequiredIfPresentContexts: make([]string, 0),
+		OptionalContexts:          make([]string, 0),
+	}, nil
```

Comment:
I don't think this is fully permissive like it is intended to be. e.g. This needs to be set to true: https://github.com/kubernetes/test-infra/blob/e4d87dae72dbf1754625cbf112afc535307c1db4/prow/config/tide.go#L41

I wonder if it would be cleaner to just implement a separate no-op/permissive `contextChecker` https://github.com/kubernetes/test-infra/blob/a1b0b85d88d615a623641ae6b01a3a9b6d2aa265/prow/tide/tide.go#L66-L71
Just an idea, setting `SkipUnknownContexts` should also work.

Reply:
I think "no-op/permissive contextChecker" is a great idea, it's a little more involved though, would prefer to do it in a separate PR. I think either way this custom logic of converting prowjobs into status contexts should be removed, so this PR can be merged first?

Reply:
forgot to mention, I have updated the struct to include `SkipUnknownContexts `

Reply:
actually implemented the interface as you suggested